punishments.js

```diff
@@ -1150,6 +1211,25 @@ Punishments.checkName = function (user, userid, registered) {
 		punishment = Punishments.userids.get(user.locked);
 		if (!punishment) punishment = ['LOCK', user.locked, 0];
 	}
+	if (!battleban && user.battlebanned) {
+		battleban = Punishments.roomUserids.get(user.battlebanned);
+		if (!battleban) battleban = ['BATTLEBAN', user.battlebanned, 0];
+	}
+	if (battleban) {
+		if (battleban[1] !== user.userid && Punishments.sharedIps.has(user.latestIp) && user.autoconfirmed) {
+			user.battlebanned = '';
+		} else {
+			Punishments.roomPunish("battle", user, battleban);
+			user.battlebanned = battleban[1];
+			user.cancelReady();
+			if (!punishment) {
+				// Prioritize popups for other global punishments
+				user.send(`|popup|Your are banned from battling${battleban[1] !== userid ? ` because you have the same IP as banned user: ${battleban[1]}` : ``}. Your battle ban will expire in a few days.${battleban[3] ? `||||Reason: ${battleban[3]}` : ``}${Config.appealurl ? `||||Or you can appeal at: ${Config.appealurl}` : ``}`);
```

Comment:
These popup messages seem to be randomly strewn across the code; they should possibly be consistent.

Also, general admonition about nesting `` ` ``

Reply:
Also "Your are"

Reply:
Heh it’s always nice when you arent the one who screwed up, will fix